src/protocol.rs

```diff
@@ -155,6 +150,20 @@ impl<P: ProtocolParticipant> Participant<P> {
             }
         };
 
+        // Check recipient
+        if message.to() != self.id {
+            Err(CallerError::WrongMessageRecipient)?
+        }
+
+        // Check that message is from a participant in this session
+        if !self
+            .participant
+            .all_participants()
+            .contains(&message.from())
+        {
+            Err(CallerError::WrongMessageRecipient)?
```

Comment:
todo: Let's make a separate error for this case so it's easier for the caller to understand what went wrong. You can add it to `CallerError` just like the others -- maybe like `InvalidMessageSender`.